applications/API/serializers.py

```
@@ -0,0 +1,572 @@
+
+        Returns:
+            Dict[str, str]: A dictionary containing 'category_key' and 'category_name'.
+
+        Raises:
+            serializers.ValidationError: If the category cannot be resolved by key or name.
+        """
+        if not value:
+            return {"category_key": "", "category_name": "Uncategorized"}
+
+        # If a model instance is already present (FK loaded), extract directly
+        from applications.API.models import InsultCategory as _IC
```

Pylint (3.3.7)

[warning] Reimport 'InsultCategory' (imported line 19)

(W0404)